src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/paymentnotification/PaymentNotificationProcessingService.java

```diff
@@ -15,12 +19,65 @@
 @Slf4j
 public class PaymentNotificationProcessingService {
 
-  public PaymentNotificationIngestionFlowFileActivityResult processPaymentNotification(
+  private final PaymentNotificationMapper paymentNotificationMapper;
+  private final PaymentNotificationErrorsArchiverService paymentNotificationErrorsArchiverService;
+
+  public PaymentNotificationProcessingService(
+      PaymentNotificationMapper paymentNotificationMapper,
+      PaymentNotificationErrorsArchiverService paymentNotificationErrorsArchiverService) {
+    this.paymentNotificationMapper = paymentNotificationMapper;
+    this.paymentNotificationErrorsArchiverService = paymentNotificationErrorsArchiverService;
+  }
+
+
+  public PaymentNotificationIngestionFlowFileResult processPaymentNotification(
       Iterator<PaymentNotificationIngestionFlowFileDTO> iterator,
       IngestionFlowFile ingestionFlowFile, Path workingDirectory) {
-      return new PaymentNotificationIngestionFlowFileActivityResult(Collections.emptyList(),0L);
+    List<PaymentNotificationErrorDTO> errorList = new ArrayList<>();
+    List<PaymentNotificationDTO> paymentNotificationList = new ArrayList<>();
+    long processedRows = 0;
+    long totalRows = 0;
+
+    while (iterator.hasNext()) {
+      totalRows++;
+
+      PaymentNotificationIngestionFlowFileDTO paymentNotificationDTO = iterator.next();
+
+      try {
+        paymentNotificationList.add(paymentNotificationMapper.map(paymentNotificationDTO, ingestionFlowFile));
+
```

Comment:
here you should save the entity calling the save API?
can we put a TODO with the JIRA ticket?

Reply:
still I don't see the invocation of the save API, am I missing something?